### PR TITLE
Migrate tahrir-api from Python2 to Python3

### PR DESCRIPTION
**Pull request by sayanchowdhury**
Signed-off-by: Sayan Chowdhury <sayan.chowdhury2012@gmail.com>

### CODE REVIEW

File: tahrir_api/dbapi.py [hunk 1]
```diff
@@ -946,18 +979,14 @@ def _adjust_ranks(self, person, old_rank):
         # own position, but needs to demote the rest).
 
         # Otherwise, take our calculations and commit them to the db.
-        for _person, data in leaderboard.items():
-            _person.rank = data['rank']
+        for _person, data in list(leaderboard.items()):
```
Comment (cverna) [inline tahrir_api/dbapi.py]: I don't think this is needed

File: tahrir_api/utils.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def convert_name_to_id(name):
 
     badge_id = name.lower().replace(" ", "-")
     bad = ['"', "'", '(', ')', '*', '&', '?']
-    replacements = dict(zip(bad, [''] * len(bad)))
-    for a, b in replacements.items():
+    replacements = dict(list(zip(bad, [''] * len(bad))))
```
Comment (cverna) [inline tahrir_api/utils.py]: I don't think we need that

File: tahrir_api/utils.py [hunk 3]
```diff
@@ -28,8 +28,8 @@ def convert_name_to_id(name):
...
+    for a, b in list(replacements.items()):
```
Comment (cverna) [inline tahrir_api/utils.py]: same here

File: tahrir_api/model.py [hunk 4]
```diff
@@ -252,12 +258,12 @@ def recipient_default(context):
     Session = sessionmaker(context.engine)()
     person_id = context.current_parameters['person_id']
     person = Session.query(Person).filter_by(id=person_id).one()
-    return unicode(hashlib.sha256(
-        person.email + context.current_parameters['salt']).hexdigest())
+    return str(hashlib.sha256((
```
Comment (cverna) [inline tahrir_api/model.py]: I don't think you need `str` hexdigest already returns a str

File: tahrir_api/model.py [hunk 5]
```diff
@@ -209,7 +215,7 @@ def __json__(self):
 
 
 def invitation_id_default(context):
-    return unicode(hashlib.md5(salt_default(context)).hexdigest())
+    return str(hashlib.md5(salt_default(context).encode('utf-8')).hexdigest())
```
Comment (cverna) [inline tahrir_api/model.py]: I don't think you need str hexdigest already returns a str

File: tahrir_api/model.py [hunk 6]
```diff
@@ -252,12 +258,12 @@ def recipient_default(context):
...
     person = Session.query(Person).filter_by(id=person_id).one()
-    return unicode(hashlib.sha256(
-        person.email + context.current_parameters['salt']).hexdigest())
+    return str(hashlib.sha256((
+        person.email + context.current_parameters['salt']).encode('utf-8')).hexdigest())
 
 
 def salt_default(context):
-    return unicode(uuid.uuid4())
+    return str(uuid.uuid4())
```
Comment (cverna) [inline tahrir_api/model.py:246]: if you return `uuid.uuid4().bytes` then you don't need to encode it here https://github.com/fedora-infra/tahrir-api/pull/46/commits/501d0692a8b4cb18d21abee3d147939a9fcdec9b#diff-bbea1d4b7899eebccc87dccb4e388c54R218

File: tahrir_api/model.py (hunk 4 above)
Comment (sayanchowdhury) [reply tahrir_api/model.py]: Yes, it does return `str`

File: tox.ini [hunk 7]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,diff-cover
+envlist = py35,py36
```
Comment (cverna) [inline tox.ini]: let's add py37

**Comment by cverna** [comment]
:fireworks: LGTM

**Review by cverna — APPROVED** [review]

**Comment by sayanchowdhury** [comment]
@cverna Thanks :100:

**Pull request merged**